modules/renter/contractor/contracts.go

```diff
@@ -553,6 +552,15 @@ func (c *Contractor) threadedContractMaintenance() {
 		return
 	}
 
+	pendingScans := c.hdb.PendingScans()
+	if pendingScans > 0 {
```

Comment:
the way that I envisioned this working was having 'RandomHosts' return an error if the hostdb itself felt that it was not at a place where it could provide a safe random sampling of hosts. That way, instead of having the contractor trying to figure out when the hostdb is safe for use, the hostdb worries about when itself is safe for use, and we separate concerns better.

That also means you don't need to export a function to count the number pending scans, you can just let the hostdb handle all of that black magic itself (in an area of the code where it's no longer black magic)